sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverClientIntegrationTest.java

```diff
@@ -0,0 +1,578 @@
+ */
+public class ServiceBusReceiverClientIntegrationTest extends IntegrationTestBase {
+
+    private final ClientLogger logger = new ClientLogger(ServiceBusReceiverClientIntegrationTest.class);
+
+    private ServiceBusReceiverClient receiver;
+    private ServiceBusSenderClient sender;
+
+    /* Sometime not all the messages are cleaned-up. This is buffer to ensure all the messages are cleaned-up.*/
+    private static final int BUFFER_MESSAGES_TO_REMOVE = 10;
+
+    /**
+     * Receiver used to clean up resources in {@link #afterTest()}.
+     */
+    private ServiceBusReceiverClient receiveAndDeleteReceiver;
+
+    final AtomicInteger messagesPending = new AtomicInteger();
+    final AtomicReference<List<Long>>  messagesDeferred = new AtomicReference<>(new ArrayList<>());
+
+
+    protected ServiceBusReceiverClientIntegrationTest() {
+        super(new ClientLogger(ServiceBusReceiverClientIntegrationTest.class));
+    }
+
+    @Override
+    protected void beforeTest() {
+        sessionId = UUID.randomUUID().toString();
+    }
+
+    @Override
+    protected void afterTest() {
+        final int pending = messagesPending.get();
+        if (pending < 1 && messagesDeferred.get().size() < 1) {
+            dispose(receiver, sender, receiveAndDeleteReceiver);
+            return;
+        }
+        // In the case that this test failed... we're going to drain the queue or subscription.
+        if (pending > 0) {
+            try {
+                IterableStream<ServiceBusReceivedMessage> removedMessage = receiveAndDeleteReceiver.receive(
+                    pending + BUFFER_MESSAGES_TO_REMOVE);
+                removedMessage.stream().forEach(receivedMessage -> {
+                    System.out.println("Removed Message sequence : " +  receivedMessage.getSequenceNumber());
+                });
+            } catch (Exception e) {
+                logger.warning("Error occurred when draining queue.", e);
+            }
+        }
+
+        if (messagesDeferred.get().size() > 0) {
+            try {
+                List<Long> listOfDeferredMessages =  messagesDeferred.get();
+                for (Long seqNumber : listOfDeferredMessages) {
+                    ServiceBusReceivedMessage deferredMessages = receiver.receiveDeferredMessage(seqNumber);
+                    receiver.complete(deferredMessages);
+                }
+            } catch (Exception e) {
+                logger.warning("Error occurred when draining deferred messages Entity : " + receiver.getEntityPath(), e);
+            }
+
+            // set empty list
+            messagesDeferred.set(new ArrayList<>());
+        }
+
+        dispose(receiver, sender, receiveAndDeleteReceiver);
+    }
+
+    /**
+     * Verifies that we can send and receive two messages.
+     */
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void receiveTwoMessagesAutoComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
```

Comment:
Auto-complete is not supported in the synchronous scenario. I would expect you to be completing these manually.